src/expression.c

```diff
@@ -2761,6 +2761,29 @@ Expression *Expression::toBoolean(Scope *sc)
             goto Lagain;
         }
     }
+    // Other non-scalar types that are valid for bool conditions
+    // should have their contents lowered to valid scalars.
+    else if (tb->ty == Tarray)
```

Comment:
I take it that the auto-tester doesn't like comments before ``else if`` conditions?